container/navigation.go

```diff
@@ -0,0 +1,118 @@
+package container
+
+import (
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/theme"
+	"fyne.io/fyne/v2/widget"
+)
+
+// Navigation container is used to provide your application with a control bar and an area for content objects.
+// Objects can be any CanvasObject, and only the most recent one will be visible.
+//
+// Since: 2.6
+type Navigation struct {
```

Comment:
The way this is written it can only be constructed via `NewNavigation` and not easily extended, or constructed by a struct literal instead, as has currently been a requirement for all builtin Fyne widgets. Most of this struct's contents may be able to move to the renderer instead.

Reply:
Good point. Have to look from that angle again and compare to other widgets.

Reply:
This needs more thought than I expected to make sure everything is initialized correctly, but I started making some parts public.

Reply:
Got it working by adding a setup method and a bunch of checks.